baremetal/metal3remediation_manager.go

```diff
@@ -289,3 +356,130 @@ func (r *RemediationManager) GetCapiMachine(ctx context.Context) (*clusterv1.Mac
 	}
 	return capiMachine, nil
 }
+
+// GetNode returns the Node associated with the machine in the current context.
+func (r *RemediationManager) GetNode(ctx context.Context) (*corev1.Node, error) {
+	capiMachine, err := r.GetCapiMachine(ctx)
+	if err != nil {
+		r.Log.Error(err, "metal3Remediation's node could not be retrieved")
+		return nil, errors.Wrapf(err, "metal3Remediation's node could not be retrieved")
+	}
+
+	clusterClient, err := r.getClusterClient(ctx)
+	if err != nil {
+		r.Log.Error(err, "Could not get cluster client")
+		return nil, errors.Wrapf(err, "Could not get cluster client")
+	}
+
+	node, err := clusterClient.Nodes().Get(ctx, capiMachine.Status.NodeRef.Name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return nil, nil
+	} else if err != nil {
+		r.Log.Error(err, "Could not get cluster node")
+		return nil, errors.Wrapf(err, "Could not get cluster node")
+	}
+	return node, nil
+}
+
+// UpdateNode updates the given node.
+func (r *RemediationManager) UpdateNode(ctx context.Context, node *corev1.Node) error {
+	clusterClient, err := r.getClusterClient(ctx)
```

Comment:
It might be more efficient to pass the client in, so that e.g. we don't have to create it twice to get and then update the Node.

Reply:
done